### PR TITLE
[Bug fix] sigmoid_bw: use sigmoid(x) * sigmoid(-x) instead of s * (1 - s)

### LINKED ISSUE #55724: [Bug] ttnn.sigmoid_bw returns a zero gradient for x >= 6.25 (bf16) / 16.75 (fp32): s * (1 - s) cancels once s rounds to 1
### Describe the bug

`ttnn.sigmoid_bw` returns a gradient of exactly `0` for every input above `6.25` in bfloat16 and `16.75` in float32, although `d/dx sigmoid(x) = sigmoid(x) * sigmoid(-x)` is an ordinary normal float out to `|x| = 87.34`, where it is still `1.2e-37`.

The composite forms the derivative as `s * (1 - s)`
([`unary_backward.cpp#L467-L480`](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp#L467-L480)):

```cpp
Tensor sig_result = ttnn::sigmoid(input, ..., SigmoidMode::ACCURATE, output_mem_config);
Tensor rsub_term = ttnn::rsub(sig_result, 1.0f, std::nullopt, output_mem_config);
Tensor prod_term_1 = ttnn::multiply(sig_result, rsub_term, std::nullopt, output_mem_config);
grad_tensor.emplace_back(ttnn::multiply(prod_term_1, grad, std::nullopt, output_mem_config));
```

`s = sigmoid(x)` tends to `1` from below and rounds to **exactly** `1` as soon as `1 - s` falls below half an ULP of `1` — below `2^-9` for bfloat16 and `2^-25` for float32. `rsub(s, 1)` is then exactly `0` and the whole gradient goes with it. The information is not in `s` any more, so no amount of accuracy in the forward `sigmoid` can recover it; the derivative has to be written without the subtraction.

The identity `sigmoid'(x) = sigmoid(x) * sigmoid(-x)` gives the same function with no cancellation: `sigmoid(-x)` *is* `1 - s`, evaluated directly.

### Repro

```python
import torch, ttnn

device = ttnn.open_device(device_id=0)
x = torch.tensor([[5.0, 10.0, 16.75, 20.0, 30.0, 60.0, 85.0]], dtype=torch.float32)
t = ttnn.from_torch(x, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
g = ttnn.from_torch(torch.ones_like(x), dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)

print(ttnn.to_torch(ttnn.sigmoid_bw(g, t)[0]))
# tensor([[6.6480e-03, 4.5357e-05, 0., 0., 0., 0., 0.]])
xd = x.double()
print(torch.sigmoid(xd) * torch.sigmoid(-xd))
# tensor([[6.6481e-03, 4.5396e-05, 5.3158e-08, 2.0612e-09, 9.3576e-14, 8.7565e-27, 1.2161e-37]])
```

### Observed vs expected

float32, gradient `1`, golden `sigmoid(x) * sigmoid(-x)` in float64. ULP is the integer distance between float32 bit patterns:

| input | got | want | ULP |
|---|---|---|---|
| 5 | 0.006648033 | 0.006648057 | 51 |
| 10 | 4.535708e-05 | 4.539581e-05 | 10,647 |
| 16.75 | **0** | 5.315785e-08 | 862,212,012 |
| 20 | **0** | 2.061154e-09 | 822,977,587 |
| 30 | **0** | 9.357623e-14 | 701,675,270 |
| 60 | **0** | 8.756511e-27 | 338,522,313 |
| 85 | **0** | 1.216099e-37 | 36,013,792 |

bfloat16, gradient `1`, ULP between bfloat16 bit patterns:

| input | got | want | ULP |
|---|---|---|---|
| 3 | 0.04467773 | 0.04516602 | 2 |
| 4 | 0.01916504 | 0.0177002 | 12 |
| 6.25 | **0** | 0.001922607 | 15,100 |
| 6.28125 | **0** | 0.001861572 | 15,092 |
| 10 | **0** | 4.529953e-05 | 14,398 |
| 40 | **0** | 4.255494e-18 | 8,861 |
| 85 | **0** | 1.219575e-37 | 550 |

The relative error over the zeroed band is 100%. The rows below the threshold show the same cancellation more mildly: at `x = 10` in float32 the gradient is already 10,647 ULP off, entirely because `1 - s` has thrown away most of its significant bits.

### Affected input range

All 65,536 bfloat16 bit patterns as the input, gradient `1`, counting inputs whose exact gradient is a normal float but whose result is exactly zero (subnormal inputs excluded):

| dtype | zeros where the golden is normal | affected range | max ULP | mean ULP |
|---|---|---|---|---|
| bfloat16 | 487 | `[6.25, 87]` | 15,100 | 80.08 |
| float32 | 297 | `[16.75, 87]` | 862,212,012 | 2,511,838 |

The gradient is symmetric, so the same information is lost on the negative side too — there it is recovered because `s` itself is accurate for `x < 0`; only the `1 - s` half of the product is destroyed.

### Which architectures

All of them. `sigmoid_bw` is a host-side composite in `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp`, so Wormhole and Blackhole build the same op graph and produce the same values. It is independent of the SFPU sigmoid kernel work in #53258 / #55061 / #54602: the loss is in the composite's algebra, not in the LUT.

### How it was verified

ttsim Blackhole v1.10.6 (`TT_METAL_SIMULATOR=libttsim_bh.so`, slow dispatch). Exhaustive sweep of all 65,536 bfloat16 bit patterns as the input with an all-ones gradient, in both bfloat16 and float32, against `sigmoid(x) * sigmoid(-x)` evaluated in float64. Subnormal inputs (flushed by the hardware) are excluded.

`test_backward_sigmoid.py::test_bw_sigmoid` does not catch it: it draws inputs from `[-1, 1]` and asserts PCC at 0.90.

sigmoid_bw before/after

### PR DESCRIPTION
**Pull request by badnikhil**
### PR Category

Bug fix

### Summary

Closes #55724

`sigmoid_bw` formed the derivative as `s * (1 - s)`. `s = sigmoid(x)` rounds to exactly `1` once `1 - s` falls below half an ULP of `1`, so `rsub(s, 1)` is exactly `0` and the gradient with it — for every input above `6.25` in bfloat16 and `16.75` in float32, where the true gradient is still a normal float out to `1.2e-37`.

`sigmoid'(x) = sigmoid(x) * sigmoid(-x)` is the same function with the subtraction removed: `sigmoid(-x)` **is** `1 - s`, evaluated directly instead of by cancellation.

All 65,536 bfloat16 bit patterns as the input with an all-ones gradient, golden `sigmoid(x) * sigmoid(-x)` in float64, on ttsim Blackhole:

| dtype | zeros where the golden is normal | max ULP | mean ULP |
|---|---|---|---|
| bfloat16 | 487 → **0** | 15,100 → **2** | 80.0844 → **0.0218** |
| float32 | 297 → **0** | 862,212,012 → **4** | 2,511,838.31 → **0.0668** |

Spot values, float32:

| input | before | after | exact |
|---|---|---|---|
| 5 | 0.006648033 | **0.006648056** | 0.006648057 |
| 10 | 4.535708e-05 | **4.539581e-05** | 4.539581e-05 |
| 16.75 | 0 | **5.315785e-08** | 5.315785e-08 |
| 30 | 0 | **9.357624e-14** | 9.357623e-14 |
| 85 | 0 | **1.216099e-37** | 1.216099e-37 |

sigmoid_bw before/after

### Notes for reviewers

- Host composite, so both Wormhole and Blackhole are covered by the one change.
- Costs one extra dispatch (`neg` plus a second `sigmoid`, replacing the `rsub`). That is the price of the identity; there is no way to recover the lost bits from `s` alone, because `s` is exactly `1.0` there.
- Both `sigmoid` calls keep `SigmoidMode::ACCURATE` and `VecMode::RC`, unchanged from before.
- Independent of the SFPU sigmoid work in #53258 / #55061 / #54602 — the loss was in the composite's algebra, not in the LUT, and the fix does not touch any kernel.
- `silu_bw` uses the same `rsub(sigmoid(x), 1)` shape but is **not** affected: there the term appears as `(1 - s) * x + 1`, which tends to `1` rather than to `0`, so cancellation there loses nothing. Left alone deliberately.
- `test_bw_sigmoid_all_bitpatterns` sweeps every bfloat16 bit pattern in both bfloat16 and float32, asserts no normal gradient comes back as zero, and holds 8 ULP. It fails both parametrisations on `main` ("487 inputs returned a zero gradient, first at x=6.25" / "297 ... first at x=16.75"). The reference is evaluated in float64 rather than through the registered golden, so the assertion is not limited by the reference's own float32 rounding.
- The pre-existing `test_bw_sigmoid` PCC cases still pass; they did not catch this because they draw inputs from `[-1, 1]` and assert PCC at 0.90.
- Verified on ttsim Blackhole v1.10.6 (slow dispatch); no silicon.

### CODE REVIEW

**Comment by mateusznowakTT** [comment]
Hi @badnikhil, there was unary_bw ops infra change (https://github.com/tenstorrent/tt-metal/pull/56061) that affected op modified in this PR, please align.
